From b4299d0bf2ffebbdcdd2e93c51a3a3951a5aba59 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Fri, 25 May 2018 18:19:47 +0800
Subject: [PATCH] arm: Makefile: pack android boot.img for arm

resource.img include logo optional.
boot.img: Image + resource.img
zboot.img: zImage + resource.img

Change-Id: I4da1ecd12200e3270f116c3fadcfa7ecbe30d7f9
Signed-off-by: Eddie Cai <eddie.cai.linux@gmail.com>
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 arch/arm/Makefile | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/arch/arm/Makefile b/arch/arm/Makefile
index a3bdfec545c2..4755ebe87896 100644
--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -382,15 +382,33 @@ endef
 
 kernel.img: zImage
 	$(Q)$(srctree)/scripts/mkkrnlimg $(objtree)/arch/arm/boot/zImage $(objtree)/kernel.img >/dev/null
-	@echo '  Image:  kernel.img is ready'
+	@echo '  Image:  kernel.img (with zImage) is ready'
 ifdef CONFIG_MODULES
 	$(Q)$(MAKE) modules
 endif
 
+ifdef CONFIG_FB
 LOGO := $(notdir $(wildcard $(srctree)/logo.bmp))
 LOGO_KERNEL := $(notdir $(wildcard $(srctree)/logo_kernel.bmp))
+endif
 %.img: %.dtb kernel.img $(LOGO) $(LOGO_KERNEL)
 	$(Q) if [ -f $(srctree)/$(LOGO) -a "$(srctree)" != "$(objtree)" ]; then cp -a $(srctree)/$(LOGO) $(objtree)/; fi
 	$(Q) if [ -f $(srctree)/$(LOGO_KERNEL) -a "$(srctree)" != "$(objtree)" ]; then cp -a $(srctree)/$(LOGO_KERNEL) $(objtree)/; fi
 	$(Q)$(srctree)/scripts/resource_tool $(objtree)/arch/arm/boot/dts/$*.dtb $(LOGO) $(LOGO_KERNEL)
 	@echo '  Image:  resource.img (with $*.dtb $(LOGO) $(LOGO_KERNEL)) is ready'
+	$(Q) if [ -x $(srctree)/scripts/mkbootimg ]; then \
+		$(srctree)/scripts/mkbootimg \
+			--base 0 --kernel_offset $(textofs-y) \
+			--kernel $(objtree)/arch/arm/boot/Image \
+			--second resource.img \
+			-o boot.img && \
+		echo '  Image:  boot.img (with Image resource.img) is ready'; \
+		$(srctree)/scripts/mkbootimg \
+			--base 0 --kernel_offset 0 \
+			--kernel $(objtree)/arch/arm/boot/zImage \
+			--second resource.img \
+			-o zboot.img && \
+		echo '  Image:  zboot.img (with zImage resource.img) is ready'; \
+	fi
+
+CLEAN_FILES += boot.img kernel.img resource.img zboot.img
-- 
2.35.3

